tests/rkt_image_list_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/steveeJ/gexpect"
+)
+
+type ImageId struct {
+	path string
+	hash string
+}
+
+func (imgId *ImageId) getShortHash(length int) (string, error) {
+	hashLen := 1 + length
+	if hashLen > len(imgId.hash) {
+		return "", fmt.Errorf("getShortHash: Hash %s is shorter than %d chars", imgId.hash, hashLen)
+	}
+
+	return imgId.hash[:hashLen], nil
+}
+
+// containsConflictingHash returns an ImageId pair if a conflicting short hash is found. The minimum
+// hash of 2 char is used for comparisons.
+func (imgId *ImageId) containsConflictingHash(imgIds []ImageId) (imgIdPair []ImageId, found bool) {
+	shortHash, err := imgId.getShortHash(2)
+	if err != nil {
+		panic(fmt.Sprintf("containsConflictingHash: %s", err))
+	}
+
+	for _, iId := range imgIds {
+		if strings.HasPrefix(iId.hash, shortHash) {
+			imgIdPair = []ImageId{*imgId, iId}
+			found = true
+			break
+		}
+	}
+	return
+}
+
+// TestShortHash tests that the short hash generated by the rkt image list
+// command is usable by the commands that accept image hashes.
+func TestShortHash(t *testing.T) {
+	var (
+		imageIds []ImageId
+		iter     int
+	)
+
+	for {
```

Comment:
Can you add a comment before the `for` to say something like:
```
// Generate images until two images with the same hash prefix are found. imageIds will contain those two images.
```
I was a bit confused that `imageIds` is used both as a temporary variable to contain lots of images, and as the variable to store the result with the 2 images.